indexdata.go

```diff
@@ -376,6 +382,30 @@ func (r *ngramIterationResults) candidates() []*candidateMatch {
 func (d *indexData) iterateNgrams(query *query.Substring) (*ngramIterationResults, error) {
 	str := query.Pattern
 
+	if len(query.Pattern) >= bloomHashMinWordLength {
+		// test against appropriate content or filename bloom filters
+		pat := []byte(query.Pattern)
+		fnMatch := false
+		contentMatch := false
+
+		if !query.Content {
+			fnMatch = d.bloomNames.maybeHasBytes(pat)
+		}
+		if !fnMatch && !query.FileName {
+			contentMatch = d.bloomContents.maybeHasBytes(pat)
+		}
+
+		if (query.FileName && !fnMatch) ||
+			(query.Content && !contentMatch) ||
+			(!fnMatch && !contentMatch) {
```

Comment:
Looking at the code below, it looks like `query.FileName` and `query.Content` are mutually exclusive. IE we look at FileName trigrams iff query.FileName, otherwise we look at content trigrams.

I believe the code you wrote here is handling them not being mutually exclusive which makes it more complicated. I think you can get away with something as simple as:

```go
var match bool
if query.FileName {
  match = d.bloomNames.maybeHasBytes(pat)
} else {
  match = d.bloomContents.maybeHasBytes(pat)
}
```

This also matches how we decide below, so not only is simpler but is using the same logic as the existing code.

Reply:
Nice simplification, done.